airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/operation/BigQueryStorageOperations.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.operation
+
+import com.google.cloud.bigquery.TableId
+import io.airbyte.cdk.integrations.destination.record_buffer.SerializableBuffer
+import io.airbyte.integrations.base.destination.operation.StorageOperations
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId
+import io.airbyte.integrations.base.destination.typing_deduping.TyperDeduperUtil
+import io.airbyte.integrations.destination.bigquery.BigQueryGcsOperations
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter
+import io.airbyte.integrations.destination.bigquery.typing_deduping.BigQueryDestinationHandler
+import io.airbyte.integrations.destination.bigquery.typing_deduping.BigQuerySqlGenerator
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.time.Instant
+import java.util.*
+
+private val log = KotlinLogging.logger {}
+class BigQueryStorageOperations(
+    private val bigQueryGcsOperations: BigQueryGcsOperations,
+    private val sqlGenerator: BigQuerySqlGenerator,
+    private val destinationHandler: BigQueryDestinationHandler
+) : StorageOperations {
+    override fun prepareStage(streamId: StreamId, destinationSyncMode: DestinationSyncMode) {
+        // Prepare staging table
+        bigQueryGcsOperations.createSchemaIfNotExists(streamId.rawNamespace)
```

Comment:
`BigQueryGcsOperations` is such a misleading name >.> can we just call it BigqueryEverythingOperations? :joy:

Reply:
Killed it down the stack.